### PR TITLE
Update user

### PR DESCRIPTION
**Pull request by Nour-Benmohamed**
want to get feedback on this. thanks!

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -119,6 +143,19 @@ export function createEvent(event, history) {
...
+        axios.put(`${ROOT_URL}/events/${eventID}`, event)
+            .catch((error) => {
+                return dispatch(authError(error.response));
+            }).then((response) => {
+                return response.data;
+            });
+    };
+}
+
+
```
Comment (seandartmouth) [inline src/actions/index.js:158]: Could you get rid of the extra space here too lol

File: src/actions/index.js [hunk 2]
```diff
@@ -147,6 +184,8 @@ export function fetchEvents(callback) {
     };
 }
 
+
+
```
Comment (seandartmouth) [inline src/actions/index.js:188]: and here

File: src/components/updateUser.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            event.stopPropagation();
+        } else {
+            const user = {
+                username,
+                first_name: firstName,
+                last_name: lastName,
+                education,
+                job,
+            };
+            this.props.createUser(user, this.props.history, this.props.editUser);
```
Comment (seandartmouth) [inline src/components/updateUser.js:40]: How is this updating the user if you're calling create user?

File: src/components/app.js [hunk 4]
```diff
@@ -13,6 +13,8 @@ import EventList from './eventlist';
 import CreateEvent from './createevent';
 import JoinEvent from './joinevent';
 import EventInfo from './eventinfo';
+import UpdateEvent from './updateEvent';
+import updateUser from './updateUser';
```
[the remark this answers is not in the document]
Comment (Nour-Benmohamed) [reply src/components/app.js:17]: done

File: src/components/app.js [hunk 5]
```diff
@@ -87,10 +89,12 @@ class App extends Component {
                         <PrivateRoute exact path="/" user={this.state.user} component={(props) => <Welcome {...props} user={this.state.user} />} />
                         <Route path="/signin" component={(props) => <SignIn user={this.state.user} editUser={this.editUser} />} />
                         <Route path="/signup" component={(props) => <SignUp user={this.state.user} editUser={this.editUser} />} />
+                        <Route path="/user/:userID" component={(props) => <UpdateUser user={this.state.user} editUser={this.editUser} />} />
                         <Route path="/about" component={About} />
                         <Route path="/event" component={(props) => <Event user={this.state.user} />} />
                         <Route path="/eventlist" component={(props) => <EventList user={this.state.user} />} />
                         <Route path="/createevent" component={(props) => <CreateEvent user={this.state.user} />} />
+                        <Route path="/joinevent/:eventID" component={(props) => <UpdateEvent user={this.state.user} />} />
```
Comment (seandartmouth) [inline src/components/app.js:97]: For UpdateUser and UpdateEvent, you should probably not pass userID and eventID in the url. Could you try to do it the way it works for eventinfo? You can see in lines 30-47 of eventlist.js how to pass parameters with a NavLink, and you can see examples of how to access these parameters that you pass in 52, 58, and 61 of eventinfo.js.

File: src/components/updateUser.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        };
+    }
+
+    handleSubmit = (event) => {
+        const form = event.currentTarget;
+        const username = document.getElementById('username').value;
+        const firstName = document.getElementById('first_name').value;
+        const lastName = document.getElementById('last_name').value;
+        const education = document.getElementById('current_education').value;
+        const job = document.getElementById('current_job').value;
```
Comment (seandartmouth) [inline src/components/updateUser.js:26]: I don't know if this is the exact right list of attributes for the People table. I think you should be updating FirstName, LastName, PersonalDescription, IndustryID, Email, and Password. I don't think we had username, education, or job in the EERD.

File: src/components/app.js [hunk 7]
```diff
@@ -87,10 +89,12 @@ class App extends Component {
                         <PrivateRoute exact path="/" user={this.state.user} component={(props) => <Welcome {...props} user={this.state.user} />} />
                         <Route path="/signin" component={(props) => <SignIn user={this.state.user} editUser={this.editUser} />} />
                         <Route path="/signup" component={(props) => <SignUp user={this.state.user} editUser={this.editUser} />} />
+                        <Route path="/user/:userID" component={(props) => <UpdateUser user={this.state.user} editUser={this.editUser} />} />
```
Comment (seandartmouth) [inline src/components/app.js:92]: Where do you navigate to UpdateUser and UpdateEvent from??

File: src/components/updateEvent.js [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+                    </Form.Row>
+                    <div className="buttons">
+                        <Button type="submit" variant="success">Submit</Button>
+                    </div>
+                </Form>
+            </div>
+        );
+    }
+}
+const mapStateToProps = (state) => (
```
Comment (a-matusewicz) [inline src/components/updateEvent.js:145]: Remove this

File: src/components/updateEvent.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+        this.state = {
+            isEditing: false,
+            validated: false,
+        };
+    }
+
+    // get the eventID to edit
+    componentDidMount = () => {
+        this.props.fetchEvent(this.props.match.params.eventID);
```
Comment (a-matusewicz) [inline src/components/updateEvent.js:23]: Pass a callback to set the result to the state like with other api calls

File: src/components/eventinfo.js [hunk 10]
```diff
@@ -2,17 +2,36 @@ import React, { Component } from 'react';
...
+            isEditing: false,
+            name: this.props.currentEvent.name,
+            time: this.props.currentEvent.time,
+            desc: this.props.currentEvent.desc,
+            ind: this.props.currentEvent.ind,
+        };
+    }
+
+    componentDidMount = () => {
+        this.props.fetchEvent(this.props.location.eventData.id);
```
Comment (a-matusewicz) [inline src/components/eventinfo.js:31]: Pass a callback to set the result to the state like with other api calls

**Pull request merged**